core/corehttp/gateway.go

```diff
@@ -6,11 +6,13 @@ import (
 	core "github.com/jbenet/go-ipfs/core"
 )
 
-func GatewayOption(n *core.IpfsNode, mux *http.ServeMux) error {
-	gateway, err := newGatewayHandler(n)
-	if err != nil {
-		return err
+func GatewayOption(writable bool) ServeOption {
+	return func(n *core.IpfsNode, mux *http.ServeMux) error {
+		gateway, err := newGatewayHandler(n, writable)
+		if err != nil {
+			return err
+		}
+		mux.Handle("/ipfs/", gateway)
```

Comment:
There's some complexity emerging in `ServeHTTP`. There may be a simpler way to handle multiple HTTP methods on the same path. If we switch over to the gorilla mux, we get the following...

```Go
state := &gatewayState{node: n, writable: writable, dir: load()}
ipfsPath := r.PathPrefix("/ipfs/").Subrouter()
ipfsPath.Methods("GET").HandlerFunc(getHandler(state))
ipfsPath.Methods("PUT").HandlerFunc(putHandler(state))
ipfsPath.Methods("POST").HandlerFunc(postHandler(state))
ipfsPath.Methods("DELETE").HandlerFunc(deleteHandler(state))
```

Thoughts?